integration-tests/src/test/groovy/com/okta/sdk/tests/it/UsersIT.groovy

```diff
@@ -176,6 +178,47 @@ class UsersIT extends ITSupport implements CrudTestSupport {
         return client.listUsers().iterator()
     }
 
+    @Test
+    @Scenario("create-user-with-user-type")
+    void createUserWithUserTypeTest() {
+
+        def password = 'Passw0rd!2@3#'
+        def firstName = 'John'
+        def lastName = 'Activate'
+        def email = "john-activate=${uniqueTestName}@example.com"
+
+        // 1. Create a User Type
+        String name = "java_sdk_user_type_" + RandomStringUtils.randomAlphanumeric(15)
+
+        UserType createdUserType = client.createUserType(client.instantiate(UserType)
+            .setName(name)
+            .setDisplayName(name)
+            .setDescription(name + "_test_description"))
+        registerForCleanup(createdUserType)
+
+        assertThat(createdUserType.getId(), notNullValue())
+
+        // 2. Create a user with the User Type
+        User user = UserBuilder.instance()
+            .setEmail(email)
+            .setFirstName(firstName)
+            .setLastName(lastName)
+            .setPassword(password.toCharArray())
+            .setActive(true)
+        // See https://developer.okta.com/docs/reference/api/user-types/#specify-the-user-type-of-a-new-user
+            .setType(client.instantiate(UserType).setId(createdUserType.getId()))
```

Comment:
nit, you could probably just pass in the `createdUserType` from above instead of creating one

Reply:
Yes, I had thought the same until I realized that we need to supply only the type `id` in the request body (in the context of creating a new user). I had added a comment in the previous line with a link to dev doc that notes this requirement.